multiraft/multiraft.go

```diff
@@ -341,6 +354,42 @@ func (s *state) start() {
 			s.stop()
 			return
 
+		case req := <-s.reqChan:
+			_, groupOK := s.groups[req.GroupID]
```

Comment:
Pull this out into a separate method instead of doing so much directly in the select/case.